app/bin/configuration.dart

```diff
@@ -50,3 +50,25 @@ class Configuration {
       : projectId = 'dartlang-pub-dev',
         packageBucketName = 'dartlang-pub-dev--pub-packages';
 }
+
+/// Configuration from the environment variables.
+class EnvConfig {
+  final String gcloudKey;
+  final String gcloudProject;
+
+  EnvConfig._({this.gcloudKey, this.gcloudProject});
+
+  factory EnvConfig._detect() {
+    return new EnvConfig._(
+      gcloudKey: Platform.environment['GCLOUD_KEY'],
```

Comment:
I would throw an exception if the `GCLOUD_PROJECT` is not set, because that is a requirement (for local and prod), the key is optional iirc.